portal-user-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/portaluser/datastore/rdbms/PortalUserDAO.java

```diff
@@ -190,7 +190,7 @@ public PreparedStatement createPreparedStatement(Connection connection) throws S
                         ps.setInt(colIdx++, portalUser.getCustomerId());
                         ps.setString(colIdx++, portalUser.getUsername());
                         ps.setString(colIdx++, portalUser.getPassword());
-                        ps.setInt(colIdx++, portalUser.getRole().getId());
+                        ps.setInt(colIdx++, portalUser.getRole().get(0).getId());
```

Comment:
Here is where it gets ugly. Our java model supports a list of roles, but the DAO silently stores only the first one. We need to fix this, and the rowMapper as well.

Reply:
The portal_user table can be changed like this:
`
create table if not exists portal_user (
    -- postgresql     
    id BIGSERIAL PRIMARY KEY,

    customerId int,
    username varchar(100),
    password varchar(300),
    role int,
    roles varchar(600),
    details bytea,
    
    createdTimestamp bigint not null,
    lastModifiedTimestamp bigint not null
  
);
`
we can add a column roles, and store the list of comma-separated role ids in it.